driver/src/price_finding/stablex_naive_solver.rs

```diff
@@ -0,0 +1,208 @@
+use crate::price_finding::error::PriceFindingError;
+use crate::price_finding::PriceFinding;
+use crate::util;
+use dfusion_core::models::{AccountState, Order, Solution};
+use std::cmp::Ordering;
+
+/// A naive solver for the stableX contract. It works almost identically to the
+/// Snapp naive solver except it has a few extra requirements and it doesn't try
+/// to be as optimal.
+///
+/// The solver works by finding the first pair of orders which matching sell and
+/// buy tokens, where one of the traded tokens is the fee token, whose users have
+/// enough balance to fully fill, and that has a viable solution.
+///
+/// The solution is computed by taking the price of the larger order and and
+/// completely filling the smaller order. A large order is defined by having
+/// more non-fee token volume. Account balances are checked for each solution.
+pub struct StableXNaiveSolver;
+
+impl PriceFinding for StableXNaiveSolver {
+    fn find_prices(
+        &self,
+        orders: &[Order],
+        accounts: &AccountState,
+    ) -> Result<Solution, PriceFindingError> {
+        Ok(ipairs(orders)
+            .filter(|&(i, j)| {
+                orders[i].buy_token == orders[j].sell_token
+                    && orders[i].sell_token == orders[j].buy_token
+            })
+            .filter(|&(i, _)| orders[i].buy_token == 0 || orders[i].sell_token == 0)
+            .filter(|&(i, j)| {
+                has_enough_balance(&orders[i], accounts) && has_enough_balance(&orders[j], accounts)
+            })
+            .filter_map(|(i, j)| compute_solution(orders, i, j))
+            .nth(0)
+            .unwrap_or_else(|| Solution::trivial(orders.len())))
+    }
+}
+
+const ETH: u128 = 1_000_000_000_000_000_000u128;
```

Comment:
I would rather name it: 
 NORMALIZED_PRICE or AMOUNT_OF_DIGITS 

I have seen that you guys like this as it is constant in Solidity, but I think it is not the best name. But I don't have a strong opinion either

Reply:
In solidity they have the `ether` unit which I find nice. I don't want to call it `NORMALIZED_PRICE` because that implies it is a measurement and not a unit as I intend it to be.